File: financial-monitor-client/src/types/transaction.ts
export const TransactionStatus = {
  Pending: "Pending",
  Completed: "Completed",
  Failed: "Failed",
} as const;

export type TransactionStatus =
  (typeof TransactionStatus)[keyof typeof TransactionStatus];

export type Transaction = {
  id: string;
  amount: number;
  currency: string;
  status: TransactionStatus;
  createdAt: string;
};


const statusMap: Record<number, TransactionStatus> = {
  0: TransactionStatus.Pending,
  1: TransactionStatus.Completed,
  2: TransactionStatus.Failed,
};

export const mapStatusToLabel = (status: string | number): TransactionStatus => {

  return statusMap[+status] ?? TransactionStatus.Pending;
};

export const normalizeTransaction = (t: any): Transaction => ({
  ...t,
  status: mapStatusToLabel(t.status),
});

File: FinancialMonitor/FinancialMonitor/Models/Transaction.cs
namespace FinancialMonitor.Models;

public class Transaction
{
    public Guid Id { get; set; } = Guid.NewGuid();

    public decimal Amount { get; set; }

    public string Currency { get; set; } = string.Empty;

    public TransactionStatus Status { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public uint xmin { get; set; }
}

public enum TransactionStatus
{
    Pending,
    Completed,
    Failed
}